src/train.py

```diff
@@ -0,0 +1,145 @@
+    "resnet50": resnet50,
+    "convnext": convnext_base,
+    "vgg16": vgg16,
+    "efficient_net_v2": efficientnet_v2_m,
+    "mobilenet_v3": mobilenet_v3_large,
+    "resnext50": resnext50_32x4d,
+    "swin": swin_b,
+    "vit": vit_b_16,
+    "unet_resnet50": smp.Unet
+    # TODO"ssd_vgg16": ssd300_vgg16,
+    # TODO "fasterrcnn_resnet50_v2": fasterrcnn_resnet50_fpn_v2,
+}
+
+
+def print_requirements():
+    pkgs = freeze.freeze()
+    for pkg in pkgs:
+        logger.info(pkg)
+
+
+def main(args):
+    if args.list_requirements:
+        print_requirements()
+
+    args_dict = vars(args)
+    logger.info(f"User Arguments {args_dict}")
+
+    dataset = InMemoryDataset(width=args.width, height=args.width)
+    data_loader = DataLoader(
+        dataset,
+        num_workers=args.n_workers,
+        batch_size=args.batch_size,
+        shuffle=False,
+        pin_memory=True,
+        drop_last=True,
+    )
+
+    trainer = Trainer(
+        accelerator=args.accelerator,
+        strategy="ddp",
+        precision=args.precision,
+        max_steps=args.n_iters + args.warmup_steps,
+        max_epochs=1,
+        callbacks=[BenchmarkCallback(warmup_steps=args.warmup_steps)],
+        devices=args.devices,
+    )
+
+    if args.model in ARCHITECTURES:
+        if args.model == "unet_resnet50":
+            model = ARCHITECTURES[args.model](
+                encoder_name="resnet50", encoder_weights=None
+            )
+        else:
+            model = ARCHITECTURES[args.model]()
+
+    else:
+        raise ValueError("Architecture not supported.")
+
+    model = LitClassification(model=model)
+    trainer.fit(model=model, train_dataloaders=data_loader)
+
+
+if __name__ == "__main__":
+    logger.info("########## STARTING NEW BENCHMARK RUN ###########")
+
+    if not torch.cuda.is_available():
+        raise ValueError("CUDA device not found on this system.")
+    else:
+        logger.info(f"CUDA Device Name: {torch.cuda.get_device_name(0)}")
+        logger.info(f"CUDNN version: {torch.backends.cudnn.version()}")
+        logger.info(
+            f"CUDA Device Total Memory: {(torch.cuda.get_device_properties(0).total_memory / 1e9):.2f} GB"
+        )
+
+    parser = argparse.ArgumentParser(description="Benchmark CV models training on GPU.")
+
+    parser.add_argument("--batch-size", type=int, required=True)
+    parser.add_argument(
+        "--n-iters",
+        type=int,
+        default=300,
+        help="Number of training iterations to benchmark for. One iteration = one batch update",
+    )
+    parser.add_argument(
+        "--precision", choices=["32", "16", "16-mixed", "bf16-mixed"], default="32"
+    )
+    parser.add_argument(
+        "--n-workers",
+        type=int,
+        default=4,
+        help="Number of Data Loader workers. CPU shouldn't be a bottleneck with 4+.",
+    )
+    parser.add_argument("--devices", type=int, default=1)
+
+    parser.add_argument("--width", type=int, default=192, help="Input width")
```

Comment:
should width and height be parameters at all? maybe default them to 224 (Imagenet values) and keep them static

Reply:
224 can be a default, but some users may have use-cases where sizes can be much bigger/smaller and the architecture performance can depend on that. 

It also doesn't hurt if this is a parameter.